### DISCUSSION TITLE
Optionally record original HASH for archives

### DISCUSSION THREAD

**Discussion by Cducharme84** [comictagger/comictagger#712] [Ideas]
It's a slightly weird idea/request for sure but the idea came about this way. I've created a script that mirrors all my metadata files into folders named after the archive in question (some managed by mylar, some custom created, lots scene based so no real consistency in file name itself) using the library's folder structure mirrored as well to keep things somewhat organized.

I did this as a personal backup, but decided well I know CV API is rough and I've got lots of data from sources that can't be automatically grepped from (think creator pages, publisher sites, previewsworld, errr fetlife yes there's a story there, etc) and for books the 3 DBs don't even recognize because it's apparently me and their mom that bought them.

However my own lack of organization is making things rough for folks and they asked about matching by hash, so I had to explain my current hashes include the very files they're looking at embedded and thus will not match.

Then I had the dim bulb turn on, "what if?" Jeffrey Wright said in my brain and boom the idea to record filename and original + new file hash for each archive written when the option is turned on into either a spreadsheet or DB table in the comictagger sqlite DB.

**Comment by lordwelch** [comment]
Putting this into the db gets messy real quick because if you use rar files (I know I'm in the minority there) you can't easily make files the same even if they have the same content (there's like 2 bytes that change no matter what) zip files we could I believe make it consistent but if it gets re-tagged the new hash would still change and paths can collide for the original name so I don't really see a good way to put this in the db that isn't a bunch of work.

So there is the scan info tag that we could put the original hash in and only add it if it doesn't have a hash. Hashing each file will also make tagging take a bit longer so that would probably be a flag to enable it at all.

**Reply by Cducharme84** [reply]
That sounds like a reasonable plan and something I can help people script when trying to use similar collection of data and sharing. The DB definitely would get messy with updated hashes, and it being a switch/argument would prevent overwriting of original hashes on updates so is actually quite optimal really.

**Reply by lordwelch** [reply]
@Cducharme84 is there a particular hash scheme you would like to be the default?
- blake2b
- blake2s
- md5
- sha1
- sha224
- sha256
- sha384
- sha3_224
- sha3_256
- sha3_384
- sha3_512
- sha512
- shake_128
- shake_256

**Reply by lordwelch** [reply]
@Cducharme84 I've now merged this in #716. I updated the format to be `sum:<hash scheme>:<hash>`

**Reply by killo3967** [reply]
Whe you talk about the hash. Is the hash of all the book/comic/file or only for the fisrt page (front page)?

**Reply by lordwelch** [reply]
Yes, it's the same as running `md5sum /path/to/comic.cbz` before writing any changes to the comic
